In [2]:
rows = list(string.ascii_uppercase[:8])
rows

['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']

In [3]:
columns = list(range(1, 13))
columns

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]

In [4]:
plate1 = pd.DataFrame('plate1', index=rows, columns=columns)
plate2 = pd.DataFrame('plate2', index=rows, columns=columns)
plate3 = pd.DataFrame('plate3', index=rows, columns=columns)
plate4 = pd.DataFrame('plate4', index=rows, columns=columns)
plate4

1       2       3       4       5       6       7       8       9   \
A  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
B  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
C  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
D  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
E  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
F  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
G  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   
H  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4  plate4   

       10      11      12  
A  plate4  plate4  plate4  
B  plate4  plate4  plate4  
C  plate4  plate4  plate4  
D  plate4  plate4  plate4  
E  plate4  plate4  plate4  
F  plate4  plate4  plate4  
G  plate4  plate4  plate4  
H  plate4  plate4  plate4

In [5]:
plates = OrderedDict([('plate1', plate1), ('plate2', plate2), ('plate3', plate3), ('plate4', plate4)  ])
plates

OrderedDict([('plate1',
                     1       2       3       4       5       6       7       8       9   \
              A  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              B  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              C  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              D  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              E  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              F  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              G  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              H  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
              
                     10      11      12  
              A  plate1  plate1  plate1  
              B  plate1  plate1  plate1  
              C 

In [6]:
plate1

1       2       3       4       5       6       7       8       9   \
A  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
B  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
C  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
D  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
E  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
F  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
G  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   
H  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1  plate1   

       10      11      12  
A  plate1  plate1  plate1  
B  plate1  plate1  plate1  
C  plate1  plate1  plate1  
D  plate1  plate1  plate1  
E  plate1  plate1  plate1  
F  plate1  plate1  plate1  
G  plate1  plate1  plate1  
H  plate1  plate1  plate1

In [7]:
plate1.unstack().reset_index().head()

level_0 level_1       0
0        1       A  plate1
1        1       B  plate1
2        1       C  plate1
3        1       D  plate1
4        1       E  plate1

In [8]:
dfs = []

for name, plate in plates.items():
    print(name)
    df = plate.unstack().reset_index()
    dfs.append(df)
tidy_plates = pd.concat(dfs, ignore_index=True)
print(tidy_plates.shape)
tidy_plates.head()

plate1
plate2
plate3
plate4
(384, 3)


level_0 level_1       0
0        1       A  plate1
1        1       B  plate1
2        1       C  plate1
3        1       D  plate1
4        1       E  plate1

In [9]:
tidy_plates.tail()

level_0 level_1       0
379       12       D  plate4
380       12       E  plate4
381       12       F  plate4
382       12       G  plate4
383       12       H  plate4

In [10]:
tidy_plates.sample(10)

level_0 level_1       0
134        5       G  plate2
257        9       B  plate3
168       10       A  plate2
243        7       D  plate3
299        2       D  plate4
351        8       H  plate4
226        5       C  plate3
295        1       H  plate4
183       11       H  plate2
239        6       H  plate3

In [11]:
tidy_plates.head()

level_0 level_1       0
0        1       A  plate1
1        1       B  plate1
2        1       C  plate1
3        1       D  plate1
4        1       E  plate1

In [12]:
tidy_plates = tidy_plates.rename(columns={'level_0': 'column_number', 'level_1': 'row_letter', 0: 'plate_value'})
tidy_plates['plate_value_letter_number'] = tidy_plates.apply(
    lambda x: '{plate_value}_{row_letter}{column_number:02}'.format(**x), axis=1
)
print(tidy_plates.shape)
tidy_plates.head()

(384, 4)


column_number row_letter plate_value plate_value_letter_number
0              1          A      plate1                plate1_A01
1              1          B      plate1                plate1_B01
2              1          C      plate1                plate1_C01
3              1          D      plate1                plate1_D01
4              1          E      plate1                plate1_E01

In [13]:
THREE_EIGHTY_FOUR_ROW_LETTERS = list(string.ascii_uppercase[:16])
THREE_EIGHTY_FOUR_ROW_LETTERS

['A',
 'B',
 'C',
 'D',
 'E',
 'F',
 'G',
 'H',
 'I',
 'J',
 'K',
 'L',
 'M',
 'N',
 'O',
 'P']

### Test out the code on a mini dataframe

In [14]:
df = tidy_plates.query("row_letter == 'H' & column_number == 12")
df

column_number row_letter plate_value plate_value_letter_number
95              12          H      plate1                plate1_H12
191             12          H      plate2                plate2_H12
287             12          H      plate3                plate3_H12
383             12          H      plate4                plate4_H12

In [15]:
df['new_column_number'] = [23, 24, 23, 24]
df['new_row_letter'] = ['O', 'O', 'P', 'P']
df

/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


column_number row_letter plate_value plate_value_letter_number  \
95              12          H      plate1                plate1_H12   
191             12          H      plate2                plate2_H12   
287             12          H      plate3                plate3_H12   
383             12          H      plate4                plate4_H12   

     new_column_number new_row_letter  
95                  23              O  
191                 24              O  
287                 23              P  
383                 24              P

In [16]:
df.pivot(index='new_row_letter', columns='new_column_number', values='plate_value')

new_column_number      23      24
new_row_letter                   
O                  plate1  plate2
P                  plate3  plate4

In [17]:
df.pivot(index='new_row_letter', columns='new_column_number', 
         values='plate_value_letter_number')

new_column_number          23          24
new_row_letter                           
O                  plate1_H12  plate2_H12
P                  plate3_H12  plate4_H12

### Test out whether it works in a for loop

In [25]:
THREE_EIGHTY_FOUR_N_ROWS = 16
THREE_EIGHTY_FOUR_N_COLUMNS = 24

for (row_letter, column_number), df in tidy_plates.groupby(['row_letter', 'column_number']):
    print(f'\n--- {row_letter}{column_number} ---')

    row_i = string.ascii_uppercase.index(row_letter)
    column_i = column_number * 2 - 1
    
    new_row_letters = string.ascii_uppercase[row_i*2:(row_i*2+2)]
    new_column_numbers = list(range(column_i, column_i+2))
#     print(f"new_row_letters: {new_row_letters}")
#     print(f"new_column_numbers: {new_column_numbers}")
    
    # Concatenate the lists of column numbers to repeat them so the 
    # numbers alternate
    # E.g. turns [1, 2] + [1, 2] --> [1, 2, 1, 2]
    df['new_column_number'] = new_column_numbers + new_column_numbers
#     print(df['new_column_number'])
    
    # Iterate to 4 and divide by 2, force it to be an integer so the 
    # decimal gets removed so we can access the first item twice 
    # and the second item twice
    # e.g. turns [A, B] --> [A, A, B, B]
    df['new_row_letter'] = [new_row_letters[int(i/2)] for i in range(4)]
#     print(df['new_row_letter'])
    print(df)

    # Pivot the data to make sure it worked to create the right "shape"
    pivoted = df.pivot(index='new_row_letter', columns='new_column_number', 
         values='plate_value_letter_number')
    print(pivoted)


--- A1 ---
     column_number row_letter plate_value plate_value_letter_number  \
0                1          A      plate1                plate1_A01   
96               1          A      plate2                plate2_A01   
192              1          A      plate3                plate3_A01   
288              1          A      plate4                plate4_A01   

     new_column_number new_row_letter  
0                    1              A  
96                   2              A  
192                  1              B  
288                  2              B  
new_column_number           1           2
new_row_letter                           
A                  plate1_A01  plate2_A01
B                  plate3_A01  plate4_A01

--- A2 ---
     column_number row_letter plate_value plate_value_letter_number  \
8                2          A      plate1                plate1_A02   
104              2          A      plate2                plate2_A02   
200              2          A      plat

/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:18: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


     column_number row_letter plate_value plate_value_letter_number  \
32               5          A      plate1                plate1_A05   
128              5          A      plate2                plate2_A05   
224              5          A      plate3                plate3_A05   
320              5          A      plate4                plate4_A05   

     new_column_number new_row_letter  
32                   9              A  
128                 10              A  
224                  9              B  
320                 10              B  
new_column_number          9           10
new_row_letter                           
A                  plate1_A05  plate2_A05
B                  plate3_A05  plate4_A05

--- A6 ---
     column_number row_letter plate_value plate_value_letter_number  \
40               6          A      plate1                plate1_A06   
136              6          A      plate2                plate2_A06   
232              6          A      plate3          

     column_number row_letter plate_value plate_value_letter_number  \
65               9          B      plate1                plate1_B09   
161              9          B      plate2                plate2_B09   
257              9          B      plate3                plate3_B09   
353              9          B      plate4                plate4_B09   

     new_column_number new_row_letter  
65                  17              C  
161                 18              C  
257                 17              D  
353                 18              D  
new_column_number          17          18
new_row_letter                           
C                  plate1_B09  plate2_B09
D                  plate3_B09  plate4_B09

--- B10 ---
     column_number row_letter plate_value plate_value_letter_number  \
73              10          B      plate1                plate1_B10   
169             10          B      plate2                plate2_B10   
265             10          B      plate3         

     column_number row_letter plate_value plate_value_letter_number  \
11               2          D      plate1                plate1_D02   
107              2          D      plate2                plate2_D02   
203              2          D      plate3                plate3_D02   
299              2          D      plate4                plate4_D02   

     new_column_number new_row_letter  
11                   3              G  
107                  4              G  
203                  3              H  
299                  4              H  
new_column_number           3           4
new_row_letter                           
G                  plate1_D02  plate2_D02
H                  plate3_D02  plate4_D02

--- D3 ---
     column_number row_letter plate_value plate_value_letter_number  \
19               3          D      plate1                plate1_D03   
115              3          D      plate2                plate2_D03   
211              3          D      plate3          

     column_number row_letter plate_value plate_value_letter_number  \
52               7          E      plate1                plate1_E07   
148              7          E      plate2                plate2_E07   
244              7          E      plate3                plate3_E07   
340              7          E      plate4                plate4_E07   

     new_column_number new_row_letter  
52                  13              I  
148                 14              I  
244                 13              J  
340                 14              J  
new_column_number          13          14
new_row_letter                           
I                  plate1_E07  plate2_E07
J                  plate3_E07  plate4_E07

--- E8 ---
     column_number row_letter plate_value plate_value_letter_number  \
60               8          E      plate1                plate1_E08   
156              8          E      plate2                plate2_E08   
252              8          E      plate3          

     column_number row_letter plate_value plate_value_letter_number  \
85              11          F      plate1                plate1_F11   
181             11          F      plate2                plate2_F11   
277             11          F      plate3                plate3_F11   
373             11          F      plate4                plate4_F11   

     new_column_number new_row_letter  
85                  21              K  
181                 22              K  
277                 21              L  
373                 22              L  
new_column_number          21          22
new_row_letter                           
K                  plate1_F11  plate2_F11
L                  plate3_F11  plate4_F11

--- F12 ---
     column_number row_letter plate_value plate_value_letter_number  \
93              12          F      plate1                plate1_F12   
189             12          F      plate2                plate2_F12   
285             12          F      plate3         

     column_number row_letter plate_value plate_value_letter_number  \
7                1          H      plate1                plate1_H01   
103              1          H      plate2                plate2_H01   
199              1          H      plate3                plate3_H01   
295              1          H      plate4                plate4_H01   

     new_column_number new_row_letter  
7                    1              O  
103                  2              O  
199                  1              P  
295                  2              P  
new_column_number           1           2
new_row_letter                           
O                  plate1_H01  plate2_H01
P                  plate3_H01  plate4_H01

--- H2 ---
     column_number row_letter plate_value plate_value_letter_number  \
15               2          H      plate1                plate1_H02   
111              2          H      plate2                plate2_H02   
207              2          H      plate3          

### Write out a separate function to add these columns to the `tidy_plates` dataframe

Sort the plates by row letter and column number so the reassignment happens in the proper order

In [82]:
tidy_plates = tidy_plates.sort_values(['row_letter', 'column_number'])
tidy_plates.head()

column_number row_letter plate_value plate_value_letter_number  \
0                1          A      plate1                plate1_A01   
96               1          A      plate2                plate2_A01   
192              1          A      plate3                plate3_A01   
288              1          A      plate4                plate4_A01   
8                2          A      plate1                plate1_A02   

    three_eighty_four_row_letter  three_eighty_four_column_number  
0                              A                                1  
96                             E                                1  
192                            I                                1  
288                            M                                1  
8                              A                                5

Assign the grouped dataframe to its own variable for safekeeping and reducing typing (which causes typos) later

In [83]:
grouped = tidy_plates.groupby(['row_letter', 'column_number'], 
                              group_keys=False)
grouped

We can check the number of items in each one to make sure the groups got formed properly

In [84]:
grouped.size()

row_letter  column_number
A           1                4
            2                4
            3                4
            4                4
            5                4
            6                4
            7                4
            8                4
            9                4
            10               4
            11               4
            12               4
B           1                4
            2                4
            3                4
            4                4
            5                4
            6                4
            7                4
            8                4
            9                4
            10               4
            11               4
            12               4
C           1                4
            2                4
            3                4
            4                4
            5                4
            6                4
                            ..
F           7

The `grouped` is a `pandas.DataFrameGroupBy` object that we can't "see" the way we would with a regular dataframe but we can apply functions to it

In [85]:
def get_384_column_number(ninety_six_column_number, verbose=False):
    original_index = ninety_six_column_number.index
    
    # This is operated on the whole mini-dataframe column number which 
    # is by definition the same since we made mini-dataframes with groupby()
    # So we just need to extract the first one, aka the 0th index
    ninety_six_column_number = ninety_six_column_number.iloc[0]
    
    column_i = ninety_six_column_number * 2 - 1
    new_column_numbers = list(range(column_i, column_i+2))
    reordered_column_numbers = new_column_numbers + new_column_numbers
    if verbose:
        print(f"converted: {ninety_six_column_number} --> {reordered_column_numbers}")
    return pd.Series(reordered_column_numbers)

grouped['column_number'].apply(get_384_column_number, verbose=True).head()

converted: 1 --> [1, 2, 1, 2]
converted: 2 --> [3, 4, 3, 4]
converted: 3 --> [5, 6, 5, 6]
converted: 4 --> [7, 8, 7, 8]
converted: 5 --> [9, 10, 9, 10]
converted: 6 --> [11, 12, 11, 12]
converted: 7 --> [13, 14, 13, 14]
converted: 8 --> [15, 16, 15, 16]
converted: 9 --> [17, 18, 17, 18]
converted: 10 --> [19, 20, 19, 20]
converted: 11 --> [21, 22, 21, 22]
converted: 12 --> [23, 24, 23, 24]
converted: 1 --> [1, 2, 1, 2]
converted: 2 --> [3, 4, 3, 4]
converted: 3 --> [5, 6, 5, 6]
converted: 4 --> [7, 8, 7, 8]
converted: 5 --> [9, 10, 9, 10]
converted: 6 --> [11, 12, 11, 12]
converted: 7 --> [13, 14, 13, 14]
converted: 8 --> [15, 16, 15, 16]
converted: 9 --> [17, 18, 17, 18]
converted: 10 --> [19, 20, 19, 20]
converted: 11 --> [21, 22, 21, 22]
converted: 12 --> [23, 24, 23, 24]
converted: 1 --> [1, 2, 1, 2]
converted: 2 --> [3, 4, 3, 4]
converted: 3 --> [5, 6, 5, 6]
converted: 4 --> [7, 8, 7, 8]
converted: 5 --> [9, 10, 9, 10]
converted: 6 --> [11, 12, 11, 12]
converted: 7 --> [13, 14, 13

row_letter  column_number   
A           1              0    1
                           1    2
                           2    1
                           3    2
            2              0    3
Name: column_number, dtype: int64

In [86]:

def get_384_row_letter(ninety_six_row_letter, verbose=False):
    # This is operated on the whole mini-dataframe row number which 
    # is by definition the same since we made mini-dataframes with groupby()
    # So we just need to extract the first one, aka the 0th index
    ninety_six_row_letter = ninety_six_row_letter.iloc[0]
    row_i = string.ascii_uppercase.index(ninety_six_row_letter)
    
    new_row_letters = string.ascii_uppercase[row_i*2:(row_i*2+2)]
    reordered_row_letters = [new_row_letters[int(i/2)] for i in range(4)]
    if verbose:
        print(f"converted: {ninety_six_row_letter} --> {reordered_row_letters}")
    return pd.Series(reordered_row_letters)
    
grouped['row_letter'].apply(get_384_row_letter, verbose=True).head()

converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: A --> ['A', 'A', 'B', 'B']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: B --> ['C', 'C', 'D', 'D']
converted: C --> ['E', 'E', 'F', 'F']
converted: C --> ['E', 'E', 'F', 'F']
converted: C

row_letter  column_number   
A           1              0    A
                           1    A
                           2    B
                           3    B
            2              0    A
Name: row_letter, dtype: object

In [87]:
tidy_plates['three_eighty_four_row_letter'] = grouped['row_letter'].apply(get_384_row_letter).values
tidy_plates['three_eighty_four_column_number'] = grouped['column_number'].apply(get_384_column_number).values
tidy_plates.head()

column_number row_letter plate_value plate_value_letter_number  \
0                1          A      plate1                plate1_A01   
96               1          A      plate2                plate2_A01   
192              1          A      plate3                plate3_A01   
288              1          A      plate4                plate4_A01   
8                2          A      plate1                plate1_A02   

    three_eighty_four_row_letter  three_eighty_four_column_number  
0                              A                                1  
96                             A                                2  
192                            B                                1  
288                            B                                2  
8                              A                                3

## make the mega 384-well plate from pivoting the four 96 well plates with the new indices

In [88]:
mega_384_plate = tidy_plates.pivot(index='three_eighty_four_row_letter', 
                               columns='three_eighty_four_column_number', 
                               values='plate_value_letter_number')
mega_384_plate

three_eighty_four_column_number          1           2           3   \
three_eighty_four_row_letter                                          
A                                plate1_A01  plate2_A01  plate1_A02   
B                                plate3_A01  plate4_A01  plate3_A02   
C                                plate1_B01  plate2_B01  plate1_B02   
D                                plate3_B01  plate4_B01  plate3_B02   
E                                plate1_C01  plate2_C01  plate1_C02   
F                                plate3_C01  plate4_C01  plate3_C02   
G                                plate1_D01  plate2_D01  plate1_D02   
H                                plate3_D01  plate4_D01  plate3_D02   
I                                plate1_E01  plate2_E01  plate1_E02   
J                                plate3_E01  plate4_E01  plate3_E02   
K                                plate1_F01  plate2_F01  plate1_F02   
L                                plate3_F01  plate4_F01  plate3_F02   
M                                plate1_G01  plate2_G01  plate1_G02   
N                                plate3_G01  plate4_G01  plate3_G02   
O                                plate1_H01  plate2_H01  plate1_H02   
P                                plate3_H01  plate4_H01  plate3_H02   

three_eighty_four_column_number          4           5           6   \
three_eighty_four_row_letter                                          
A                                plate2_A02  plate1_A03  plate2_A03   
B                                plate4_A02  plate3_A03  plate4_A03   
C                                plate2_B02  plate1_B03  plate2_B03   
D                                plate4_B02  plate3_B03  plate4_B03   
E                                plate2_C02  plate1_C03  plate2_C03   
F                                plate4_C02  plate3_C03  plate4_C03   
G                                plate2_D02  plate1_D03  plate2_D03   
H                                plate4_D02  plate3_D03  plate4_D03   
I                                plate2_E02  plate1_E03  plate2_E03   
J                                plate4_E02  plate3_E03  plate4_E03   
K                                plate2_F02  plate1_F03  plate2_F03   
L                                plate4_F02  plate3_F03  plate4_F03   
M                                plate2_G02  plate1_G03  plate2_G03   
N                                plate4_G02  plate3_G03  plate4_G03   
O                                plate2_H02  plate1_H03  plate2_H03   
P                                plate4_H02  plate3_H03  plate4_H03   

three_eighty_four_column_number          7           8           9   \
three_eighty_four_row_letter                                          
A                                plate1_A04  plate2_A04  plate1_A05   
B                                plate3_A04  plate4_A04  plate3_A05   
C                                plate1_B04  plate2_B04  plate1_B05   
D                                plate3_B04  plate4_B04  plate3_B05   
E                                plate1_C04  plate2_C04  plate1_C05   
F                                plate3_C04  plate4_C04  plate3_C05   
G                                plate1_D04  plate2_D04  plate1_D05   
H                                plate3_D04  plate4_D04  plate3_D05   
I                                plate1_E04  plate2_E04  plate1_E05   
J                                plate3_E04  plate4_E04  plate3_E05   
K                                plate1_F04  plate2_F04  plate1_F05   
L                                plate3_F04  plate4_F04  plate3_F05   
M                                plate1_G04  plate2_G04  plate1_G05   
N                                plate3_G04  plate4_G04  plate3_G05   
O                                plate1_H04  plate2_H04  plate1_H05   
P                                plate3_H04  plate4_H04  plate3_H05   

three_eighty_four_column_number          10     ...              15  \
three_eighty_four_row_letter                    ...                   
A                    

Check one row letter and column number from the 96 well plate into the 384 well plate

In [96]:
tidy_plates.query("row_letter == 'E' & column_number == 5")

column_number row_letter plate_value plate_value_letter_number  \
36               5          E      plate1                plate1_E05   
132              5          E      plate2                plate2_E05   
228              5          E      plate3                plate3_E05   
324              5          E      plate4                plate4_E05   

    three_eighty_four_row_letter  three_eighty_four_column_number  
36                             I                                9  
132                            I                               10  
228                            J                                9  
324                            J                               10